common/containers/Tabs/RestoreKeystore/components/KeystoreDetails.tsx

```diff
@@ -0,0 +1,156 @@
+
+interface State {
+  secretKey: string;
+  password: string;
+  fileName: string;
+  isPasswordVisible: boolean;
+  isPrivateKeyVisible: boolean;
+  wallet: IFullWallet | null | undefined;
+}
+const initialState: State = {
+  secretKey: '',
+  password: '',
+  isPasswordVisible: false,
+  isPrivateKeyVisible: false,
+  fileName: '',
+  wallet: null
+};
+
+class KeystoreDetails extends Component<{}, State> {
+  public state = initialState;
+
+  public componentWillUnmount() {
+    this.resetState();
+  }
+  public render() {
+    const {
+      secretKey,
+      isPasswordVisible,
+      isPrivateKeyVisible,
+      password,
+      wallet,
+      fileName
+    } = this.state;
+    const passwordValid = password.length > 9;
+    const privateKeyValid =
+      secretKey.length === 64 && /^[a-fA-F0-9]+$/g.test(secretKey);
+    const content = (
+      <div className="KeystoreDetails">
+        <div>
+          <label className="KeystoreDetails-key">
+            <h4 className="KeystoreDetails-label">Private Key</h4>
+            <div className="input-group">
+              <input
+                className={`form-control is-${privateKeyValid
+                  ? 'valid'
+                  : 'invalid'}`}
+                type={isPrivateKeyVisible ? 'text' : 'password'}
+                name="secretKey"
+                value={secretKey}
+                onChange={this.handleInput}
+              />
+              <span
+                onClick={this.togglePrivateKey}
+                role="button"
+                className="input-group-addon eye"
+              />
+            </div>
+          </label>
+        </div>
+        <div>
+          <label className="KeystoreDetails-password">
+            <h4 className="KeystoreDetails-label">Password</h4>
+            <div className="input-group">
+              <input
+                className={`form-control is-${passwordValid
+                  ? 'valid'
+                  : 'invalid'}`}
+                type={isPasswordVisible ? 'text' : 'password'}
+                name="password"
+                value={password}
+                onChange={this.handleInput}
+              />
+              <span
+                onClick={this.togglePassword}
+                role="button"
+                className="input-group-addon eye"
+              />
+            </div>
+          </label>
+        </div>
+        {!wallet ? (
+          <button
+            onClick={this.handleKeystoreGeneration}
+            className="KeystoreDetails-submit btn btn-block"
+            disabled={!passwordValid || !privateKeyValid}
+          >
+            Generate Keystore
+          </button>
+        ) : (
+          <a
+            onClick={this.resetState}
+            href={this.getBlob()}
+            className="KeystoreDetails-download btn btn-primary btn-block"
+            aria-label="Download Keystore File (UTC / JSON · Recommended · Encrypted)"
+            aria-describedby="x_KeystoreDesc"
```

Comment:
You'll need to import the `translate` function from `import translate from 'translations';` to get the value of `x_keystoreDesc` I believe, looks like generateWallet doesn't do that either, would be a good time to fix both